sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
+1 on the introduction of a distinct Date type, but this demonstrates why (at least for dates), the prior use of `.UTC()` was not just presentational. We represent dates as timestamps whose hour/minute/second components are all zero, but those components are only defined with respect to some timezone. At least for dates, we must associate a timezone with the `time.Time` instead of leaving it to be interpreted according to the local time (or we adjust the UTC date timestamp provided by the server for the local time zone)

I'm not sure what this means for timestamps. It doesn't seem quite right for the driver to set them to display in UTC, but it's also weird to have them inconsistent with dates. I'd probably leave them in UTC for now until we have a more coherent plan for how we're going to handle non-UTC timezones (including #2327). I suspect that the solution will involve server-side awareness of the client's timezone, so the driver will just be reflecting a timezone provided by the server instead of making these decisions on its own.